library/std/src/panicking.rs

```diff
@@ -744,19 +745,22 @@ fn rust_panic_with_hook(
     if let Some(must_abort) = must_abort {
         match must_abort {
             panic_count::MustAbort::PanicInHook => {
-                // Don't try to print the message in this case
-                // - perhaps that is causing the recursive panics.
+                // Don't try to format the message in this case, perhaps that is causing the
+                // recursive panics. However if the message is just a string, no user-defined
+                // code is involved in printing it, so that is risk-free.
+                let msg_str = message.and_then(|m| m.as_str()).map(|m| [m]);
```

Comment:
A possible alternative would be to explicitly say something like "(panic message omitted as that would require formatting)" instead of silently dropping the entire message. Would that be preferred?

Reply:
I think just omitting it for now is fine. The location should be sufficient to help people find the cause of the panic.